walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -33,6 +33,40 @@ private static boolean isAttended(Event event, MeetingRequest request) {
   }
 
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> freeTimes = new ArrayList<TimeRange>(); // Output list for possible meeting times.
+    Event[] eventsArray = events.toArray(new Event[events.size()]);
+    long duration = request.getDuration() / 30; // Meeting duration in number of half hour array slots.
+    
+    if (duration > 47) return freeTimes; // If the duration is out of bounds, we don't need to add any times.
```

Comment:
You are only checking bound in one direction here. Since this is a public interface you might want to make sure your inputs are valid.

Reply:
Checked the opposite bound with this commit b9cc433